scripts/fixate-requirements.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python
+
+import os
+import os.path
+import sys
+from distutils.version import StrictVersion
+
+
+cwd = os.path.abspath(os.getcwd())
+FIXED_REQF = os.path.join(cwd, '../fixed-requirements.txt')
+LOCAL_REQF = os.path.join(cwd, 'requirements.txt')
+GET_PIP = "    curl https://bootstrap.pypa.io/get-pip.py | python"
+
+try:
+    import pip
+    from pip.req import parse_requirements
+except ImportError:
+    print "Download pip:\n", GET_PIP
+    sys.exit(1)
+
+
+def check_pip_version():
+    if StrictVersion(pip.__version__) < StrictVersion('7.0.0'):
+        print "Upgrade pip, your version `{}' "\
+              "is outdated:\n".format(pip.__version__), GET_PIP
+        sys.exit(1)
+
+
+def check_requiremnts_sources():
+    for f in (FIXED_REQF, LOCAL_REQF):
```

Comment:
Thanks. According to some PEP? Just creating list and converting it to
tuple doesn't seem to be more efficient anyway..

On Tue, Jun 23, 2015, 07:57 Lakshmi Kannan <notifications@github.com> wrote:

> In scripts/fixate-requirements.py
> <https://github.com/StackStorm/st2/pull/1647#discussion_r33009435>:
>
> > +    import pip
> > +    from pip.req import parse_requirements
> > +except ImportError:
> > +    print "Download pip:\n", GET_PIP
> > +    sys.exit(1)
> > +
> > +
> > +def check_pip_version():
> > +    if StrictVersion(pip.__version__) < StrictVersion('7.0.0'):
> > +        print "Upgrade pip, your version `{}' "\
> > +              "is outdated:\n".format(pip.__version__), GET_PIP
> > +        sys.exit(1)
> > +
> > +
> > +def check_requiremnts_sources():
> > +    for f in (FIXED_REQF, LOCAL_REQF):
>
> Nit: Best to use for f in [FIXED_REQF, LOCAL_REQF]. () is for tuple.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/StackStorm/st2/pull/1647/files#r33009435>.
>